pandas/core/indexes/base.py

```diff
@@ -4109,6 +4109,14 @@ def _assert_can_do_op(self, value):
         if not is_scalar(value):
             raise TypeError(f"'value' must be a scalar, passed: {type(value).__name__}")
 
+    @property
+    def _has_complex_internals(self):
+        """
+        Indicates if an index is not directly backed by a numpy array
+        """
+        # used to disable groupby tricks
```

Comment:
"tricks" -> "going through libreduction fastpath which would ..."?

Reply:
Copied that from the original implementation but agree it's not very helpful. Updated to something that's more informative.